### PR TITLE
Fix: Fix creation bugs to make slug unique.

### PR DESCRIPTION
**Pull request by StrongMonkey**
This PR fixes creating assistance where slug has to be unique. It makes sure user can create assistance with the same name.

It also fixes other bugs where page is unnecessarily rendered that causes other network traffic.

Ignore all the formatter issues, i am going to have a separate PR after conference to add prettier and eslint config to the project.

https://github.com/gptscript-ai/desktop/issues/141

### CODE REVIEW

File: actions/me/scripts.tsx [hunk 1]
```diff
@@ -114,16 +113,3 @@ export async function getScriptContent(scriptURL: string): Promise<string | unde
         return undefined;
     }
 }
-
-export async function getNewScriptName() {
```
Comment (StrongMonkey) [inline actions/me/scripts.tsx:118]: The logic of magically delecting assistant name is removed. Right now it defaults to `New Assistant` where user can have multiple.

File: components/scripts/create.tsx [hunk 2]
```diff
@@ -17,22 +17,22 @@ export default function Create() {
...
-                    content: newDefaultAssistant(name),
-                }).then((script) => {
-                    getScript(`${script.id}`)
-                        .then((script) =>
-                            window.location.href = `/edit?file=${script?.publicURL}&id=${script?.id}`
-                        );
-                })
-            })
+        const defaultName = "New Assistant";
+        const slug = defaultName.toLowerCase().replace(" ", "-") + "-" + Math.random().toString(36).substring(2, 7);
```
Comment (StrongMonkey) [inline components/scripts/create.tsx:21]: The logic of creating a unique slug per assistant by appending random 5 charaters.

File: contexts/edit.tsx [hunk 3]
```diff
@@ -168,11 +168,22 @@ const EditContextProvider: React.FC<EditContextProps> = ({scriptPath, children})
...
-            }).catch((error) => console.error(error));
+            if (scriptId && visibility && tools.length > 0 && root) {
+                const existing = await getScript(scriptId.toString())
+                let toUpdate: Script = {
+                    visibility: visibility,
+                    content: await stringify([root, ...tools]),
+                    id: scriptId,
+                }
+                // Only update slug when displayName has changed
+                if (existing?.displayName !== root.name) {
```
Comment (StrongMonkey) [inline contexts/edit.tsx:179]: In here we only update the slug when displayName has changed.

File: components/script.tsx [hunk 4]
```diff
@@ -99,9 +101,11 @@ const Script: React.FC<ScriptProps> = ({ className, messagesHeight = 'h-full', e
 
     return (
         <div className={`h-full w-full ${className}`}>
+            {showAssistantName && <h1
```
Comment (StrongMonkey) [inline components/script.tsx:104]: Added the title where chat can have assitant name on the top

image

**Review by thedadams — APPROVED** [review]

**Review by drpebcak — APPROVED** [review]

**Pull request merged**